cdk-infra/utils/ssm.ts

```diff
@@ -0,0 +1,128 @@
+import { GetParameterCommand, SSMClient } from '@aws-sdk/client-ssm';
+
+export function getSsmPathPrefix(env: string): string {
+  return `/env/${env}/docs/worker_pool`;
+}
+
+/**
+ * Returns the secure strings from SSM using the SSM client.
+ * @param ssmPrefix the path prefix that should contain the environment for the SSM strings (e.g. /env/dotcomstg/worker_pool/)
+ * @returns The map of environment variables.
+ */
+async function getSecureStrings(
+  ssmPrefix: string,
+  secureStrings: readonly string[],
+  paramToEnvMap: Map<string, string>,
+  resourceName: string
+) {
+  const ssmClient = new SSMClient({ region: process.env.CDK_DEFAULT_REGION });
+
+  const secureStringsMap: Record<string, string> = {};
+
+  await Promise.all(
+    secureStrings.map(async (paramName: string) => {
```

Comment:
I'm not explicitly returning anything, but I want to use map here still so I can use `Promise.all` on the resulting promises that are returned from the map. I don't believe `Promise.all` would work on a `forEach` since `forEach` doesn't return anything, even if the callback function is async.